libp2p/utility.nim

```diff
@@ -65,3 +65,16 @@ when defined(libp2p_agents_metrics):
   const
     KnownLibP2PAgents* {.strdefine.} = "nim-libp2p"
     KnownLibP2PAgentsSeq* = KnownLibP2PAgents.safeToLowerAscii().tryGet().split(",")
+
+template tryException*(body: untyped): untyped =
```

Comment:
Needs a more descriptive name
exceptionToAssert? Idk

Reply:
I didn't like the name either, thanks for the suggestion.